lib/Runtime/Library/JavascriptAsyncFunction.cpp

```diff
@@ -48,6 +48,14 @@ Var JavascriptAsyncFunction::EntryAsyncFunctionImplementation(
     auto* asyncFn = VarTo<JavascriptAsyncFunction>(function);
     auto* scriptFn = asyncFn->GetGeneratorVirtualScriptFunction();
     auto* generator = library->CreateGenerator(args, scriptFn, library->GetNull());
+
+    return BeginAsyncFunctionExecution(generator);
+}
+
+JavascriptPromise* JavascriptAsyncFunction::BeginAsyncFunctionExecution(JavascriptGenerator* generator)
+{
+    auto* library = generator->GetLibrary();
```

Comment:
Async function implementation split in two - as second half which actually executes the function can be reused for modules with Top level Await, whilst first half is not needed for modules.